libs/blocks/merch/merch.js

```diff
@@ -415,24 +422,8 @@ export async function getPriceContext(el, params) {
   const displayRecurrence = params.get('term');
   const displayTax = params.get('tax');
   const forceTaxExclusive = params.get('exclusive');
-  let template = PRICE_TEMPLATE_REGULAR;
-  // This mapping also supports legacy OST links
-  switch (params.get('type')) {
-    case PRICE_TEMPLATE_DISCOUNT:
-    case 'priceDiscount':
-      template = PRICE_TEMPLATE_DISCOUNT;
-      break;
-    case PRICE_TEMPLATE_OPTICAL:
-    case 'priceOptical':
-      template = PRICE_TEMPLATE_OPTICAL;
-      break;
-    case PRICE_TEMPLATE_STRIKETHROUGH:
-    case 'priceStrikethrough':
-      template = PRICE_TEMPLATE_STRIKETHROUGH;
-      break;
-    default:
-      break;
-  }
+  // The PRICE_TEMPLATE_MAPPING supports legacy OST links
+  const template = PRICE_TEMPLATE_MAPPING.get(params.get('type')) ?? params.get('type');
```

Comment:
this is not exactly the same. we were defaulting to PRICE_TEMPLATE_REGULAR before, and we are not anymore

Reply:
yes, i found this behaviour quite unexpected actually.
are you sure its a good idea to default 'invalid' template to 'price'?
i can revert it back but i can't think of when would it be a good idea

Reply:
i can't tell you for sure until it's an actual bug :) 
it's just that behaviour change makes me nervous